src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,21 @@
+package duke.commands;
+
+import duke.exception.DukeException;
+
+/**
+ * Abstract class of commands.
+ *
+ * @param <T>
+ */
+public abstract class Command<T> {
+    private void execute() throws DukeException, DukeException {}
```

Comment:
Any reason why you did not make it a public abstract method to be overridden by subclasses since all subclasses implements execute() ?